pkg/thread/monitor.go

```diff
@@ -0,0 +1,76 @@
+package thread
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/openshift/assisted-service/pkg/leader"
+
+	"github.com/sirupsen/logrus"
+)
+
+type Monitor struct {
```

Comment:
Monitor should have a context that can disable the operation.
Sometimes monitor can run for a long time, and we need the option to stop it in the middle.
Currently Stop() will not cancel current run and just wait for it to finish.